### ISSUE TITLE
Provider com.beust.klaxon.KlaxonJsonProvider not found

### ISSUE THREAD

**Issue by smallufo** [cbeust/klaxon#119]
I am using `com.beust.klaxon.Parser` with `com.github.kittinunf.fuel.Fuel`  for parsing json documents. It works fine in test mode. But when I compact the project to WAR file and deploy to tomcat server , it throws exception below  :  

```
javax.servlet.ServletException: org.glassfish.jersey.server.ContainerException: java.util.ServiceConfigurationError: javax.json.spi.JsonProvider: Provider com.beust.klaxon.KlaxonJsonProvider not found
	org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:489)
	org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:427)
	org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:388)
	org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:341)
	org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:228)
	org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	org.springframework.orm.jpa.support.OpenEntityManagerInViewFilter.doFilterInternal(OpenEntityManagerInViewFilter.java:178)
	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	org.springframework.orm.jpa.support.OpenEntityManagerInViewFilter.doFilterInternal(OpenEntityManagerInViewFilter.java:178)
	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
```

I searched and found `KlaxonJsonProvider` is empty : [https://github.com/cbeust/klaxon/blob/master/src/main/java/com/beust/klaxon/KlaxonJsonProvider.kt](url)

I don't know who `registered` `KlaxonJsonProvider` to jersey . 
I cannot find `KlaxonJsonProvider` in my `WEB-INF/lib` folder  : 
No such class , even no such string.

```
~/tomcat/webapps/destiny/WEB-INF/lib $ grep -e KlaxonJson * -R
Binary file klaxon-2.1.8.jar matches
~/tomcat/webapps/destiny/WEB-INF/lib $ grep -e KlaxonJsonProvider * -R
~/tomcat/webapps/destiny/WEB-INF/lib $ 
``` 

Any ideas ?

related POMs  :  

```xml
<kotlin.version>1.2.21</kotlin.version>
<kotlin.compiler.jvmTarget>1.8</kotlin.compiler.jvmTarget>

    <dependency>
      <groupId>com.github.kittinunf.fuel</groupId>
      <artifactId>fuel</artifactId>
      <version>1.12.1</version>
    </dependency>
    <dependency>
      <groupId>com.beust</groupId>
      <artifactId>klaxon</artifactId>
      <version>2.1.8</version>
    </dependency>
```

Tomcat 8.5.11

**Comment by cbeust** [comment]
No idea what's going on :-)

I just went through the [Fuel project](https://github.com/kittinunf/Fuel)  and there are no references to `KlaxonJsonProvider` anywhere.

Any thoughts who else might be trying to reference that nonexistent class?

I could possibly restore it but I need to know who is trying to use it and how.

**Comment by smallufo** [comment]
OK , I found where goes wrong .
The server runs JDK9 , the exception is : 
```
java.util.ServiceConfigurationError: javax.json.spi.JsonProvider: Provider com.beust.klaxon.KlaxonJsonProvider not found
	java.base/java.util.ServiceLoader.fail(ServiceLoader.java:588)
	java.base/java.util.ServiceLoader.access$200(ServiceLoader.java:390)
	java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1206)
	java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1215)
	java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1259)
	java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1294)
	java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1379)
	javax.json.spi.JsonProvider.provider(JsonProvider.java:89)
	javax.json.Json.createObjectBuilder(Json.java:266)
```

After searching , here is related question : https://stackoverflow.com/questions/49050419/simple-serviceloader-java-util-serviceconfigurationerror-provider-not


And I found the class string after extracting `klaxon-2.1.8.jar` 

After extracting `klaxon-2.1.8.jar` , there is a file named `javax.json.spi.JsonProvider` inside `META-INF/services` , the content is 
`com.beust.klaxon.KlaxonJsonProvider` 


I think it should be removed.

Thanks.

**Comment by cbeust** [comment]
Good catch!

Fixed in 2.1.9.
